miqa/core/rest/experiment.py

```diff
@@ -79,12 +80,18 @@ def create(self, request, *args, **kwargs):
         project = Project.objects.get(id=serializer.data['project'])
         if not get_perms(request.user, project):
             Response(status=status.HTTP_401_UNAUTHORIZED)
-        experiment = Experiment(
-            project=project,
-            name=serializer.data['name'],
-            lock_owner=None,
-        )
-        experiment.save()
+        try:
+            experiment = Experiment(
+                project=project,
+                name=serializer.data['name'],
+                lock_owner=None,
+            )
+            experiment.save()
+        except utils.IntegrityError:
+            raise APIException('An experiment with the same name already exists in this project.')
```

Comment:
I think we should call `full_clean` on the model and let the first-class validation logic form its own errors. This logic isn't very future proof; imagine if we add other check constraints or database validations to the model, then we don't know that an IntegrityError implies a name collision.

Reply:
Given that we are already calling `is_valid` above, we could add this validation logic into the serializer itself.

Reply:
Ok, I like that better. I'll make that change

Reply:
Changed in [2aee6e4](https://github.com/OpenImaging/miqa/pull/412/commits/2aee6e45b75d78842ce9b9543e2a3f8c486330e1)